federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -142,24 +171,69 @@ func NewIngressController(client federation_release_1_4.Interface) *IngressContr
 
 		&util.ClusterLifecycleHandlerFuncs{
 			ClusterAvailable: func(cluster *federation_api.Cluster) {
-				// When new cluster becomes available process all the ingresses again.
-				ic.clusterDeliverer.DeliverAt(allClustersKey, nil, time.Now().Add(ic.clusterAvailableDelay))
+				// When new cluster becomes available process all the ingresses again, and configure it's ingress controller's configmap with the correct UID
+				ic.clusterDeliverer.DeliverAfter(cluster.Name, cluster, ic.clusterAvailableDelay)
 			},
 		},
 	)
 
-	// Federated updater along with Create/Update/Delete operations.
-	ic.federatedUpdater = util.NewFederatedUpdater(ic.ingressFederatedInformer,
+	// Federated informer on configmaps for ingress controllers in members of the federation.
+	ic.configMapFederatedInformer = util.NewFederatedInformer(
+		client,
+		func(cluster *federation_api.Cluster, targetClient kube_release_1_4.Interface) (cache.Store, framework.ControllerInterface) {
+			glog.V(4).Infof("Returning new informer for cluster %q", cluster.Name)
+			return framework.NewInformer(
+				&cache.ListWatch{
+					ListFunc: func(options api.ListOptions) (pkg_runtime.Object, error) {
+						if targetClient == nil {
+							glog.Errorf("targetClient is nil")
+						}
+						return targetClient.Core().ConfigMaps(uidConfigMapNamespace).List(options) // we only want to list one by name - unfortunately Kubernetes don't have a selector for that.
```

Comment:
Yes, I'll do that.  I was initially struggling with incompatible return types, but that's easy to fix.